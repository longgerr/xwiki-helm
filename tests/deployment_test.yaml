suite: test deployment
templates:
  - deployment.yaml
tests:
  - it: should use custom image
    set:
      image.tag: latest
      mysql.enabled: false
      postgresql.enabled: false
      image.name: default
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: default:latest
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: default-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -latest          
  - it: should use mysql lts image
    set:
      image.tag: latest
      mysql.enabled: true
      postgresql.enabled: false
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-mysql-tomcat
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: xwiki-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -lts-mysql-tomcat
  - it: should use postgres lts image
    set:
      image.tag: latest
      mysql.enabled: false
      postgresql.enabled: true
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].image
          value: xwiki:lts-postgres-tomcat
      - matchSnapshot:
          path: spec.template.spec.containers[0].env
      - matchRegex:
          path: metadata.labels.app
          pattern: xwiki-*
      - matchRegex:
          path: metadata.labels.app
          pattern: -lts-postgres-tomcat 
  - it: should properly set value
    set:
      image.pullPolicy: IfNotPresent
      service.internalPort: 8080
      persistentStorage.enabled: true
    asserts:  
      - equal:
          path: metadata.name
          value: RELEASE-NAME-xwiki
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
      - equal:
          path: spec.template.spec.containers[0].ports[0].containerPort
          value: 8080
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[0].name
          value: xwiki-data
      - equal:
          path: spec.template.spec.volumes[0].name
          value: xwiki-data
      - equal:
          path: spec.template.spec.volumes[0].persistentVolumeClaim.claimName
          value: RELEASE-NAME-xwiki
      - matchSnapshot:
          path: spec.template.spec.containers[0].volumeMounts
      - matchSnapshot:
          path: spec.template.spec.volumes 
  - it: should properly set value pvc
    set:
      persistentStorage.enabled: true
    asserts:
      - isKind:
          of: PersistentVolumeClaim
        documentIndex: 1
      - matchSnapshot:
          path: spec
        documentIndex: 1
      - equal:
          path: metadata.name
          value: RELEASE-NAME-xwiki
        documentIndex: 1
  - it: should use empty local storage
    set:
      persistentStorage.enabled: false
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[0].name
          value: xwiki-data
      - equal:
          path: spec.template.spec.volumes[0].name
          value: xwiki-data
      - isEmpty:
          path: spec.template.spec.volumes[0].emptyDir
  - it: should run in node sticky mode and use hostPath storage
    set:
      persistentStorage.enabled: false
      nodeStickyMode:
        enabled: true
        dataHostPath: /app/xwiki/data
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: xwiki
                operator: In
                values:
                - "true"
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.volumes[0].hostPath.path
          value: /app/xwiki/data
      - equal:
          path: spec.template.spec.volumes[0].hostPath.type
          value: DirectoryOrCreate
      - matchSnapshot:
          path: spec.template.spec.affinity
  - it: should not run in node sticky mode even nodeAffinity is set
    set:
      persistentStorage.enabled: true
      nodeStickyMode:
        enabled: false
        dataHostPath: /app/xwiki/data
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: xwiki
                operator: In
                values:
                - "true"
    asserts:
      - isKind:
          of: Deployment
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[0].name
          value: xwiki-data
      - equal:
          path: spec.template.spec.volumes[0].name
          value: xwiki-data
      - equal:
          path: spec.template.spec.volumes[0].persistentVolumeClaim.claimName
          value: RELEASE-NAME-xwiki
      - isEmpty:
          path: spec.template.spec.affinity
  - it: should set environment variables from external DB configMap and secret
    set:
      mysql.enabled: false
      postgresql.enabled: false
      externalDB:
        password:
          value: ''
          valueFrom:
            secretKeyRef:
              name: xwiki-db-secret
              key: DB_PASSWORD_FROM_SECRET
        user:
          value: ''
          valueFrom:
            configMapKeyRef:
              name: xwiki-db-config-map
              key: DB_USER_FROM_CONFIGMAP
        database:
          value: ''
          valueFrom:
            configMapKeyRef:
              name: xwiki-db-config-map
              key: DB_NAME_FROM_CONFIGMAP
        host:
          value: ''
          valueFrom:
            configMapKeyRef:
              name: xwiki-db-config-map
              key: DB_HOST_FROM_CONFIGMAP
    asserts:
    - matchSnapshot:
          path: spec.template.spec.containers[0].env
          