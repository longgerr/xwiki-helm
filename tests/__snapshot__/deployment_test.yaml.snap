should generate additional environment variables:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: DB_PASSWORD
          name: RELEASE-NAME-xwiki
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
    - name: SOME_ENV
      value: some-value
should generate initContainer to patch config files:
  1: |
    - command:
      - /bin/bash
      - -c
      - |-
        update-config.sh -c "xwiki.cfg" -a "xwiki.authentication.authclass" -v "org.xwiki.contrib.authentication.customfield.CustomFieldAuthenticator" -r "/usr/local/xwiki/data"
        update-config.sh -c "xwiki.cfg" -a "xwiki.superadminpassword" -v "the-pass-word" -r "/usr/local/xwiki/data"
      image: langyb/xwiki-toolbox:latest
      imagePullPolicy: Always
      name: update-config
      volumeMounts:
      - mountPath: /usr/local/xwiki/data
        name: xwiki-data
should properly set value:
  1: |
    - mountPath: /usr/local/xwiki/data
      name: xwiki-data
  2: |
    - name: xwiki-data
      persistentVolumeClaim:
        claimName: RELEASE-NAME-xwiki
should properly set value pvc:
  1: |
    accessModes:
    - ReadWriteMany
    resources:
      requests:
        storage: 200Gi
should run in node sticky mode and use hostPath storage:
  1: |
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: xwiki
            operator: In
            values:
            - "true"
should set environment variables from external DB configMap and secret:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: DB_PASSWORD_FROM_SECRET
          name: xwiki-db-secret
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST_FROM_CONFIGMAP
          name: xwiki-db-config-map
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER_FROM_CONFIGMAP
          name: xwiki-db-config-map
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_NAME_FROM_CONFIGMAP
          name: xwiki-db-config-map
should use custom image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: DB_PASSWORD
          name: RELEASE-NAME-xwiki
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use mysql lts image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: mysql-password
          name: RELEASE-NAME-mysql
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
should use postgres lts image:
  1: |
    - name: DB_PASSWORD
      valueFrom:
        secretKeyRef:
          key: postgresql-password
          name: RELEASE-NAME-postgresql
    - name: INDEX_HOST
      valueFrom:
        configMapKeyRef:
          key: INDEX_HOST
          name: RELEASE-NAME-xwiki
    - name: INDEX_PORT
      valueFrom:
        configMapKeyRef:
          key: INDEX_PORT
          name: RELEASE-NAME-xwiki
    - name: DB_HOST
      valueFrom:
        configMapKeyRef:
          key: DB_HOST
          name: RELEASE-NAME-xwiki
    - name: DB_USER
      valueFrom:
        configMapKeyRef:
          key: DB_USER
          name: RELEASE-NAME-xwiki
    - name: DB_DATABASE
      valueFrom:
        configMapKeyRef:
          key: DB_DATABASE
          name: RELEASE-NAME-xwiki
